### ISSUE TITLE
Roundcube My Mail not working after 3.4 update & hotfix - then downloading update in Module Manager

### ISSUE THREAD

**Issue by PercyP** [YetiForceCompany/YetiForceCRM#3653]
Hi,

I thought I had got this working (see https://github.com/YetiForceCompany/YetiForceCRM/issues/3649#issuecomment-253158857)
bt unfortunately it has broken again. Here are the steps I followed:

1. Upgraqded 3.2 to 3.3
2. pgraded 3.3 to 3.4
3. Installed hotfix
4. Opened My Mail in the CRM menu - see Roundcube login screen
5. Go to Configuration - Email - Roundcube Settings to change the default Roundcube email settings to my own server
6. Save the settings
7. Go to My Mailbox in the CRM menu
8. Produces a 404 error inside crm window, but menus etc. still display
8. Check google chrome errors which show:
GET http://www.mysite.com/yetiforcecrm/modules/OSSMail/roundcube/?_autologin=1&_autologinKey=7e3709650270ef9c9eb5e3cf0596586b 404 (Page not found)
9. Went to Configuration - Email - Autologin and took the tick out of Autologin checkbox
10. Checked Google Chrome inspect and get 404 error: 
http://www.mysite.com/yetiforcecrm/modules/OSSMail/roundcube/ Failed to load resource: the server responded with a status of 404 (Page not found)

I tried exactly the same process on another domain - produced exactly the same result. It seems to go wrong after clicking on the Update button under Modules-installation.

I am at a loss how to fix this. Why would it fail when it was working perfectly before running the Update button?

Percy

**Comment by kpaulaha** [comment]
I have done exact the same procedures for the installation that is in our own use. It works fine, no problem as far as we have tested under 2 days, at least mail works ok.

**Comment by PercyP** [comment]
I have now got the signin box to show. It turned out to be folder permissions. For some reason my VPS doesn't like the roundcube being remotely installed. To get around it I had to use ftp and change permissions on files to 644 and directories to 755. 

Now I am back to an old problem where it comes up with the Connection to the storage server failed - even though all my mail settings are exactly as they were before all of this.

**Comment by mariuszkrzaczkowski** [comment]
Check logs without this we can not help
https://yetiforce.com/pl/dokumentacja-programisty/silnik/109-debugowanie.html

**Comment by PercyP** [comment]
Hi @mariuszkrzaczkowski 

I have gone through the tutorial and found this error in the php error log is this what you mean?

`[12-Oct-2016 19:20:51 Europe/London] PHP Warning:  stream_socket_client(): Failed to enable crypto in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858
[12-Oct-2016 19:20:51 Europe/London] PHP Warning:  stream_socket_client(): unable to connect to ssl://imap.mysite.com:993 (Unknown error) in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858
[12-Oct-2016 19:21:14 Europe/London] PHP Warning:  stream_socket_client(): Peer certificate CN=`544556.vps-10.com' did not match expected CN=`imap.mysite.com' in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858
[12-Oct-2016 19:21:14 Europe/London] PHP Warning:  stream_socket_client(): Failed to enable crypto in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858
[12-Oct-2016 19:21:14 Europe/London] PHP Warning:  stream_socket_client(): unable to connect to ssl://imap.mysite.com:993 (Unknown error) in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858`

Is that what you mean?

**Comment by mariuszkrzaczkowski** [comment]
Here's the error:
`[12-Oct-2016 19:21:14 Europe/London] PHP Warning: stream_socket_client(): Peer certificate CN=544556.vps-10.com' did not match expected CN=imap.mysite.com' in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858`
Try to set this parameter in config\modules\OSSMail.php

```
$config['imap_conn_options'] = [
	'ssl' => [
		'verify_peer' => false,
		'verfify_peer_name' => false,
		'CN_match' => '*.mysite.com',
	],
];
$config['smtp_timeout'] = 5;
$config['smtp_conn_options'] = [
	'ssl' => [
		'verify_peer' => false,
		'verfify_peer_name' => false,
		'CN_match' => '*.mysite.com',
	],
];
```

**Comment by PercyP** [comment]
@mariuszkrzaczkowski 

I tried adding those values but it still doesn't work. I am thinking it may be because I have upgraded to php 5.6 (it worked perfectly on 5.5 but yetiforce now states we need 5.6 as a minimum). I have been digging around and wondered if it has anything to do with SSL certificates and if I install one will this rectify it? Apparently 5.6 introduced some changes earlier in the year that checks for this?? and it can cause those errors - I don't know enough about this to know if it is true or not.

**Comment by mariuszkrzaczkowski** [comment]
Check logs

**Comment by PercyP** [comment]
@mariuszkrzaczkowski 

This is what the log says:

`[13-Oct-2016 12:14:04 Europe/London] PHP Warning:  stream_socket_client(): Peer certificate CN=`544556.vps-10.com' did not match expected CN=`*.mysite.com' in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858
[13-Oct-2016 12:14:04 Europe/London] PHP Warning:  stream_socket_client(): Failed to enable crypto in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858
[13-Oct-2016 12:14:04 Europe/London] PHP Warning:  stream_socket_client(): unable to connect to ssl://imap.mysite.com:993 (Unknown error) in /home/mysite/public_html/crm/modules/OSSMail/roundcube/program/lib/Roundcube/rcube_imap_generic.php on line 858`

**Comment by PercyP** [comment]
@mariuszkrzaczkowski 

Finally solved it - thanks to you :) in my case the settings needed to be:

'verify_peer' => true,
		'verify_peer_name' => true,

Thank you

**Comment by serbiaserbia** [comment]
@PercyP 
hey, I am glad your installation finally works! :-) Your resistance, knowlage and YF team expeience....
How did you resolve and transfer your customisation from old installation? Did you transfer anything or you must manually enter everything again? I mean added new fields, workflows, reports etc.

I get cooled down on thinking to manually make enter everything again. :-(

**Comment by PercyP** [comment]
@serbiaserbia 

I stuck with it as didn't relish the thought of rebuilding everything. Here are all the different things I did - they may not all be relevent but something in this might be the missing thing for you:

1. Checked the System configuration in yetiforce for any errors in php etc. and then addressed them. For example, I had an error showing for Exif so I updated my EasyApache and php making sure Exif was checked in exhaustive options - same also for LDAP (I am on a VPS so I could make these changes myself).

2. Updated my php.ini file with the recommended settings provided on yetiforce website

3. Deleted my current copy of my crm and installed the safe copy I made before upgrading from 3.2

4. Installed the upgrade from 3.2 - 3.3

5. Installed the upgrade from 3.3 to 3.4

6. Installed the hotfix then went to the module manager and downloaded each of the 4 modules including roundcube (you will get a success message for each module)

7. Went to filezilla and opened my site then used the change permissions for my CRM folder - setting all folders to 755 and all files to 644

8. Opened config/mondules/OSSMail/Rondcube/ossmail.php and changed the following (note addyoursite.com would be your own domain name:

'ssl' => [
		'verify_peer' => true,
		'verify_peer_name' => true,
'CN_match' => 'yoursitename.com',
	],
];
$config['smtp_timeout'] = 5;
$config['smtp_conn_options'] = [
	'ssl' => [
		'verify_peer' => true,
		'verify_peer_name' => true,
'CN_match' => 'yoursitename.com',
	],

**MAKE SURE PEER SET TO TRUE**

Set up mail settings for roundcube inside crm and all should work
